dart/lib/src/sentry_exception_factory.dart

```diff
@@ -0,0 +1,32 @@
+import 'protocol/sentry_stack_trace.dart';
+import 'protocol.dart';
+import 'sentry_stack_trace_factory.dart';
+
+/// class to convert Dart Error and exception to SentryException
+class SentryExceptionFactory {
+  final SentryStackTraceFactory _stacktraceFactory;
+
+  SentryExceptionFactory({SentryStackTraceFactory stacktraceFactory})
```

Comment:
curiosity, why do we use optional param if this is required?
would not be the proper way just `SentryExceptionFactory(SentryStackTraceFactory stacktraceFactory)` ?

Reply:
✅  it's now an optional argument with a default value